### PR TITLE
Fix Fixnum warning on Ruby 2.4+

### PR DESCRIPTION
**Pull request by matthiasr**
By patching Integer instead of Fixnum.

To support this change, add simple specs for the Float and Fixnum/Integer monkey-patching.

To support the specs, add rspec rake tasks.

To make the Rakefile go, add the missing development dependencies, update rspec, and update all other versions according to their constraints.

### CODE REVIEW

File: support/ruby/collins-shell/lib/collins_shell/monkeypatch.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ def to_human_size
   end
 end
 
-class Fixnum
+class Integer
```
Comment (grobie) [inline support/ruby/collins-shell/lib/collins_shell/monkeypatch.rb:18]: IIRC in ruby 2.4 they reorganized the whole class hierarchy  and changed how Integer works. Does this still work with ruby <2.4 and ruby >=2.4?

File: support/ruby/collins-shell/Rakefile [hunk 2]
```diff
@@ -62,3 +62,9 @@ YARD::Rake::YardocTask.new do |t|
   t.files = ['lib/**/*.rb']
   t.options = ['--markup', 'markdown']
 end
+
+begin
+  require 'rspec/core/rake_task'
+  RSpec::Core::RakeTask.new(:spec)
+rescue LoadError
```
Comment (grobie) [inline support/ruby/collins-shell/Rakefile]: Why is rspec the only gem in this file guarded in such way? How is it possible to reach this step given the bundler error handling on top?

**Review by grobie — APPROVED** [review]
Some questions, but generally go for it.

File: support/ruby/collins-shell/lib/collins_shell/monkeypatch.rb (hunk 1 above)
Comment (matthiasr) [reply support/ruby/collins-shell/lib/collins_shell/monkeypatch.rb:18]: I tested with 1.9.3 and 2.4.1, both do the right thing – on 2.4 this is just _the_ class, on 1.9.3 we implement this method in the parent class so it's available for both Fixnum and Bignum.

File: support/ruby/collins-shell/Rakefile (hunk 2 above)
Comment (matthiasr) [reply support/ruby/collins-shell/Rakefile]: I just copied this from the rspec documentation. I'll remove the guard.

**Pull request merged**